### PR TITLE
fix: unable to submit after fixing validation errors

### PR DESCRIPTION
**Pull request by schottra**
# TL;DR
There was a missing state transition between the invalid inputs state and the submission/validation state, causing the state machine to do nothing when attempting to submit after fixing validation errors. The fix just required adding an allowed state transition and corresponding unit tests.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?
 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
_N/A_

## Tracking Issue
https://github.com/lyft/flyte/issues/565

### CODE REVIEW

**Review by bnsblue — APPROVED** [review]
rubberstampping it cuz I don't know Typescript
